### ISSUE TITLE
Lint/RedundantRequireStatement should not remove requiring `pp`

### ISSUE THREAD

**Issue by mvz** [rubocop/rubocop#11099]
Lint/RedundantRequireStatement autocorrects `require "pp"` by removing it. However, `pp` is not loaded by default:

```bash
$ ruby -e 'puts $LOADED_FEATURES' | grep pp
/home/matijs/.rbenv/versions/3.1.2/lib/ruby/3.1.0/rubygems/path_support.rb
$ ruby --version
ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-linux]
```

## Expected behavior

RuboCop leaves `require "pp"` in place.

## Actual behavior

RuboCop removes `require "pp"`.

## Steps to reproduce the problem

- Create a file `foo.rb` containing

```
require "pp"
```

- Run `rubocop -a foo.rb --only Lint/RedundantRequireStatement`.

## RuboCop version

```
$ rubocop -V
1.37.0 (using Parser 3.1.2.1, rubocop-ast 1.22.0, running on ruby 3.1.2) [x86_64-linux]
```

**Comment by koic** [comment]
This is a quote from the official release article of Ruby 2.5.

> One of our most loved libraries, pp.rb, is now automatically loaded. You no longer have to write require "pp". [[Feature #14123]](https://bugs.ruby-lang.org/issues/14123)

https://www.ruby-lang.org/en/news/2017/12/25/ruby-2-5-0-released/

So, `require 'pp'` seems unnecessary. Are there any use cases where a problem occurs?

**Comment by mvz** [comment]
Oh, I see it auto-requires `pp` when needed. That explains why it's not in the list of loaded features.

The problem I'm facing is failing tests in this pull request: https://github.com/mvz/live_ast/pull/70

However, I think this actually exposed an issue in that gem: It tries to open the file where `pp` is defined, and fails because it's defined in the prelude.

So this is not really a good reason to remove `pp` from the list of redundant requires.

**Comment by bobf** [comment]
@koic I don't know if my use case justifies removing this cop but please see the following:

```console
$ ruby -e 'require "stringio"; io = StringIO.new; pp("foo", io); puts io.string'
"foo"                                                       
#<StringIO:0x00007f2f25e0b338>   
```

```console
$ ruby -e 'require "stringio"; io = StringIO.new; PP.pp("foo", io); puts io.string'
-e:1:in `<main>': uninitialized constant PP (NameError)
```

```console
$ ruby -e 'require "stringio"; require "pp"; io = StringIO.new; PP.pp("foo", io); puts io.string'
"foo"
```

`PP.pp` receives an optional argument which can be a `StringIO` which will receive the output, whereas `pp` pretty-prints `*args` instead. In my use case I need to store the output so I must use `PP.pp`, which depends on `require 'pp'` so I currently have disabled this cop for that line. I can live with it but wanted to see if you think the cop is behaving correctly, given how most people will use `pp` (or maybe you know a better way to store the output from `pp`).

(Tell me if you want me to create a new issue for this).

**Comment by mvz** [comment]
@bobf can you please open a new issue for this, since your use case is very different from mine?

**Comment by koic** [comment]
FYI, No need to open a new issue because I've opened #11931 to fix the false positive for `PP.pp`. Thank you.
